aten/src/ATen/templates/TensorBody.h

```diff
@@ -599,6 +599,21 @@ class CAFFE2_API Tensor {
     return impl_->grad();
   }
 
+  // The Forward AD API functions below are low level and are not to be used by end
+  // users who should use the API provided in torch/csrc/autograd.h
```

Comment:
If these really are so internal, shouldn't they (1) have leading underscores and (2) consider putting them only on TensorImpl.h and not on Tensor.h?

Reply:
Yes adding to TODO for follow up PR.